emscripten/ARToolKitJS.cpp

```diff
@@ -122,19 +141,19 @@ extern "C" {
 							if( flag == -1 ) { // Take the first or best result.
 	                flag = i;
 	                err = kpmResult[i].error;
-									ARLOGe("error in the tracking");
+									ARLOGe("error in the tracking: %d \n", err);
 	            }
 	        }
         }
-				flag = kpmResultNum;
-				ARLOGi("flag is: %d\n", flag);
-        if (flag > -1) {
+
+				//ARLOGi("flag is: %d\n", arc->detectedPage);
+        if (pageNo >= 0 && pageNo == arc->detectedPage) {
```

Comment:
the problem is this we do not pass any data to the EM_ASM 
if i change the variable in the conditional for example:
if (pageNo >= 0)  we get data but with that Threejs error, and the sphere appear and disappears. MAybe we should use another condition or what else?

Reply:
don't remember why i put this:
`if (pageNo >= 0 && pageNo == arc->detectedPage) {`

Reply:
maybe it should be

```js
if (arc->detectedPage >= -1) {

//// or better

if (arc->detectedPage > -1) {
```

Reply:
I would rather ask why appears and disappears. As you put it, it is a hack that might work for some but not all cases. I would prefer to understand why pageNo should be as condition. However, this might not have sense: `pageNo >= 0 && pageNo == arc->detectedPage`. I can see some contradictory behaviour there.

If you leave it as `arc->detectedPage > -1`, does it mean that `pageNo >= 0`? For what I read from the code, `detectedPage` was a sort of flag for different states. What meant -1 status for detectedPage?

Reply:
What about:
```
if (pageNo >= 0 && arc->detectedPage SOMETHING -1) {
```

Reply:
This is the flagging I found for detectedPage:
```
int detectedPage  = -2;  // -2 Tracking not inited, -1 tracking inited OK, >= 0 tracking online on page.
```